### PR TITLE
feat(movies-fetch): Add fetching/filtering of movies

### PR DESCRIPTION
**Pull request by mpiercy827**
What: Add GET /movies endpoint with ability to filter.

Why: Allow users to fetch and filter movies from the API.

Details:

- Allow user to fetch all movies, or filter by title and/or year.
- Fuzzy title matching implemented with Postgres fuzzystrmatch extension.
- Includes unit tests and integration tests.

### CODE REVIEW

File: lib/models/movie.js [hunk 1]
```diff
@@ -7,9 +7,40 @@ module.exports = Bookshelf.Model.extend({
   serialize: function () {
     return {
       id: this.get('id'),
-      title: this.get('title'),
+      name: this.get('name'),
```
Comment (peterdolan) [inline lib/models/movie.js]: Isn't title more appropriate here? Why the change?

File: test/models/movie.test.js [hunk 2]
```diff
@@ -11,12 +18,89 @@ describe('movie model', () => {
...
         'release_year',
         'object'
       ]);
     });
 
   });
 
+  describe('filter', () => {
+
+    before(() => {
```
Comment (peterdolan) [inline test/models/movie.test.js]: While it doesn't matter for this particular example, it's better practice to use a beforeEach loop.

beforeEach(() => {
...

File: lib/models/movie.js (hunk 1 above)
Comment (peterdolan) [inline lib/models/movie.js]: why the name change here?

File: test/models/movie.test.js [hunk 3]
```diff
@@ -11,12 +18,89 @@ describe('movie model', () => {
...
+
+      return Movie.filter(query)
+      .fetchAll()
+      .then((movies) => {
+        expect(movies.size()).to.eql(1);
+        expect(movies.at(0).get('name')).to.eql('The Matrix');
+      });
+    });
+
+    it('filters by min year', () => {
```
Comment (peterdolan) [inline test/models/movie.test.js]: More robust test might be closer to exact -- if you made min year 2012, would it still return value?

File: test/plugins/features/movies/index.test.js [hunk 4]
```diff
@@ -20,4 +20,19 @@ describe('movies integration', () => {
...
+  describe('fetch', () => {
+
+    it('fetches movies', () => {
+      return Movies.inject({
+        url: '/movies',
+        method: 'GET'
+      })
+      .then((response) => {
+        expect(response.statusCode).to.eql(200);
+        expect(response.result).to.be.an('array');
```
Comment (peterdolan) [inline test/plugins/features/movies/index.test.js]: probably better to assert length here --

expect(response.result).to.have.length(3)

File: lib/models/movie.js (hunk 1 above)
Comment (mpiercy827) [inline lib/models/movie.js]: I changed the name of the column in the zero-downtime migrations portion of miyagi

Comment (mpiercy827) [inline lib/models/movie.js]: Same as above!

Comment (peterdolan) [inline lib/models/movie.js]: they didn't have that yet when I did it!

File: test/factories/movie.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+
+module.exports = Factory.define('movie')
+  .sequence('id', (i) => `${i}`)
+  .attr('name', 'Movie')
```
Comment (peterdolan) [inline test/factories/movie.js]: For both here and the option below I don't think it makes sense to have a default -- what do you think?

Comment (mpiercy827) [inline test/factories/movie.js]: I think that makes sense. Changed!
